### ISSUE TITLE
AutoDMG Cache Builder asking for Auth?

### ISSUE THREAD

**Issue by bhuddleston** [facebook/IT-CPE#72]
Trying to get "AutoDMG Cache Builder" working. Have a new machine setup with Munki and it is connecting to our repo without any problems, but when I run the cache builder I get this error:
```
$ sudo ./autodmg_cache_build.py
Password:
Using plistlib
Something went wrong! cannot import name MunkiDownloadError
 
Using Munki repo: https://munki.ourdomain.com/munki_repo
Error: HTTPS was used but no auth provided.
```
Note we use SSL, but we don't use Basic Authentication.  So I'm not sure what auth it is asking for.  Tried to google and ask others but no success yet.

**Comment by bhuddleston** [comment]
I found these lines in the script:
```
  if "https" in MUNKI_URL and not BASIC_AUTH:
    print >> sys.stderr, "Error: HTTPS was used but no auth provided."
    sys.exit(2)
```
Which assumes that if we have HTTPS then we must have Basic Auth setup, which is false.  I can just comment out those lines, but I'm not sure what the `Something went wrong! cannot import name MunkiDownloadError` means either.

**Comment by bhuddleston** [comment]
RE: MunkiDownloadError

I talked to the MacAdmins group and it turns out I was running their in-progress release.  

I was told:

> MunkiDownloadError is definitely in fetch.py in the latest release
> But it's gone from later work-in-progress builds
> https://github.com/munki/munki/blob/master/code/client/munkilib/fetch.py#L99-L136
> 
> So I'd recommend that you go back to the 2.8.2 release
> 
> And warn @nick.mcspadden that his code will need to be adjusted for a future release

**Comment by natewalck** [comment]
Is this still and issue or is it resolved?

**Comment by bhuddleston** [comment]
The "Error: HTTPS was used but no auth provided."  is an issue still (I had to comment it out of your script).  It assumes that if someone is using HTTPS, then they must be using AUTH.  All of our web traffic is setup to go over HTTPS by default, but we currently have no plan in setting up AUTH for Munki.  

Is that a needed / required check to exit out from in your script?

**Comment by nmcspadden** [comment]
That can be safely removed, it's not needed. It was a very long time ago and I never updated it since. You could even do it over HTTP - that was mostly a sanity check to make sure we didn't have misconfigured machines in our environment.

Development on ADCB is mostly halted until we see further evidence that will continue to play a significant part in Mac deployment.

**Comment by nmcspadden** [comment]
#138 updates the version on Github with our internal copy, which (among many other things) removes those lines.

**Comment by nmcspadden** [comment]
Closed by #138
